azure-pipelines.yml

```diff
@@ -1,19 +1,72 @@
-# Starter pipeline
-# Start with a minimal pipeline that you can customize to build and deploy your code.
-# Add steps that build, run tests, deploy, and more:
-# https://aka.ms/yaml
+variables:
+  dockerVersion: '19.03.5'
+  dotnetVersion: '3.1.x'
 
 trigger:
 - master
+- development
+
+pr:
+- master
+- development
 
 pool:
   vmImage: 'ubuntu-latest'
 
 steps:
-- script: echo Hello, world!
-  displayName: 'Run a one-line script'
+
+- task: DockerInstaller@0
+  inputs:
+    dockerVersion: $(dockerVersion)
+  displayName: 'Docker: Install - $(dockerVersion)'
+
+- task: UseDotNet@2
+  displayName: 'Use DotNet Core $(dotnetVersion)'
+  inputs:
+    packageType: 'sdk'
+    version: $(dotnetVersion)
+    includePreviewVersions: false
+
+- script: |
+    docker-compose -f "docker-compose.integration.yml" up --build -d
+  displayName: 'Docker-compose up'
+
+- task: DotNetCoreCLI@2
+  displayName: 'Run unit tests'
+  inputs:
+    command: test
+    projects: '**/*API.Tests/*.csproj'
+    arguments: '-v n  --collect "Code coverage"'
+
+- task: DotNetCoreCLI@2
+  displayName: 'Run Integration tests'
+  inputs:
+    command: test
+    projects: '**/*Integration.Tests/*.csproj'
+    arguments: '-v n'
+
+- task: DockerCompose@0
+  displayName: 'Build: Dev'
+  inputs:
+      containerregistrytype: 'Azure Container Registry'
+      azureSubscription: 'NHSAPP-BuyingCatalogue (Non-Prod)'
+      azureContainerRegistry: '{"loginServer":"gpitfuturesdevacr.azurecr.io", "id" : "/subscriptions/7b12a8a2-f06f-456f-b6f9-aa2d92e0b2ec/resourceGroups/gpitfutures-dev-rg-acr/providers/Microsoft.ContainerRegistry/registries/gpitfuturesdevacr"}'
+      dockerComposeFile: '**/docker-compose-build.yml'
+      action: 'Build services'
+      additionalImageTags: '$(Build.BuildNumber)'
+      includeSourceTags: true
+
+- task: DockerCompose@0
+  displayName: 'Push: Dev'
+  inputs:
```

Comment:
you might want a condition to stop pushing to the ACR during a pull request, overwriting the image
something like
`condition: and(succeeded(), not(eq(variables['build.reason'], 'PullRequest')))`

Reply:
or use `ne` to trim that second condition down

Reply:
👍 Good idea, will do that.